ocs_sample_library_preview/SharedStreams.py

```diff
@@ -0,0 +1,482 @@
+        url=self.__stream_path.format(
+                namespace_id=namespace_id,
+                tenant_id=tenant_id,
+                stream_id=self.__base_client.encode(stream_id))
+
+        return self.__streams.getValueUrl(url, index, value_class, headers)
+
+    def getFirstValue(self, tenant_id: str, namespace_id: str, community_id: str, stream_id: str, value_class: type = None) -> Any:
+        """
+        Retrieves JSON object from Sds Service the first value to be added to
+            the stream specified by 'stream_id'
+        :param tenant_id: tenant to work against
+        :param namespace_id: namespace to work against
+        :param community_id: community to work against
+        :param stream_id: id of the stream
+        :param value_class: use this to cast the value into a given type.
+            Type must support .fromJson()  Default is None.
+            If None returns a dynamic Python object from the data.
+        :return: the value.  If value_class is defined it is in this type.
+            Otherwise it is a dynamic Python object
+        """
+        self.__validateParameters(tenant_id, namespace_id, community_id, stream_id)
+        headers = self.__base_client.communityHeaders(community_id=community_id)
+        url=self.__stream_path.format(
+                tenant_id=tenant_id,
+                namespace_id=namespace_id,
+                community_id=community_id,
+                stream_id=self.__base_client.encode(stream_id))
+
+        return self.__streams.getFirstValueUrl(url, value_class, headers)
+
+    def getLastValue(self, tenant_id: str, namespace_id: str, community_id: str, stream_id: str, value_class: type = None) -> Any:
+        """
+        Retrieves JSON object from Sds Service the last value to be added to
+            the stream specified by 'stream_id'
+        :param tenant_id: tenant to work against
+        :param namespace_id: namespace to work against
+        :param community_id: community to work against
+        :param stream_id: id of the stream
+        :param value_class: use this to cast the value into a given type.
+            Type must support .fromJson()  Default is None.
+            If None returns a dynamic Python object from the data.
+        :return: the value.  If value_class is defined it is in this type.
+            Otherwise it is a dynamic Python object
+        """
+        self.__validateParameters(tenant_id, namespace_id, community_id, stream_id)
+        headers = self.__base_client.communityHeaders(community_id=community_id)
+        url = self.__stream_path.format(
+            tenant_id=tenant_id,
+            namespace_id=namespace_id,
+            stream_id=self.__base_client.encode(stream_id))
+
+        return self.__streams.getLastValueUrl(url, value_class, headers=headers)
+
+    def getWindowValues(self, tenant_id: str, namespace_id: str, community_id: str, stream_id: str, start: str, end: str, value_class: type, filter: str = '') -> list[Any]:
```

Comment:
for every function from here down, the value_class parameter is required. I think these should be initialized to None since they're rarely used, and None is the default behavior. My last PR was to change this for the getWindowValues/URL functions in streams, so this brings up a few thoughts to me:

- This branch was copied from main before those changes were put in, which means it's not a direct copy paste anymore. It's not a merge conflict so I didn't notice it at first
- When calling the underlying __streams functions, the arguments are all positional and not named
  - Right now, this means that getWindowValues here will actually fail when run against the current main branch because the value_class parameter had to be moved behind all the required parameters. This call as is is passing None as a start time, start as an end time, and end as a value_class.
  - In general, these changes in the future will be harder to implement if there's a bunch of wrapper classes like this with unnamed arguments being passed down 
- My PR was only for the WindowValues and WindowValuesURL calls, but there's other like summaries, range, index collection, sampled, etc that have the value_class as a required parameter

Reply:
@derekendres what would you suggest here? I think we should either roll back the previous changes I made to streams.getWindowValues to make value_class a required parameter again and force users to always specify 'None' if they don't want to use it, or else make a large effort to change all of the value_class parameters here and in streams.py to use them (and optionally start naming the arguments when passing them down into an underlying class, like SharedStreams -> Streams so that future changes to things like the Streams class functions' parameter lists don't unknowingly break SharedStreams's argument list)

Reply:
i generally think named arguments is safest.  I don't like moving things around and potentially breaking code, but i think that is safest to assume that it may happen.  

i don't think anyone uses the value_class functionality (could be entirely off-base as I am not privy to everything people do), its a neat thing, but not very pythonic IMO,  With that said I don't think we should be requiring it and obviously that means we would have to default to None.  Probably could pull that out to its own WI to make sure there is consistency across the library

Reply:
Based on our discussion in stand up, I will make sure that all methods use named arguments in the SharedStreams class when calling the Streams class, and use optional value_class arguments default to None in SharedStreams as well.

Are we ok with that and to then to create a separate WI to address the change to make value_class default to None in Streams? 

Thanks for the feedback!